enhancements/console/samples.md

```diff
@@ -0,0 +1,309 @@
+---
+title: samples
+authors:
+  - "@jerolimov"
+reviewers:
+  - "@vikram-raj"
+  - "@spadgett"
+  - "@JoelSpeed"
+approvers:
+  - "@spadgett"
+api-approvers:
+  - "@JoelSpeed"
+creation-date: 2023-05-09
+last-updated: 2023-06-21
+tracking-link:
+  - https://issues.redhat.com/browse/ODC-7241
+see-also:
+  - "/enhancements/console/quick-starts.md"
+---
+
+# Samples
+
+## Summary
+
+The Developer Console contains a sample list that helps users initialize workloads and play with features like the "Import from Git" flow.
+
+Until 4.13, the [samples-operator](https://github.com/openshift/cluster-samples-operator) and the external Devfile registry ([registry.devfile.io](https://registry.devfile.io/viewer?types=sample)) provide these samples.
+
+A `ConsoleSample` CRD will allow other teams, operators, and customers to add additional samples to the console.
+
+## Background info
+
+The samples are one of the main entry points to create a Kubernetes resource from the Developer perspective:
+
+![Screenshot of the add page that shows a lot of option, incl. an option to "View all samples"](samples-add-page.png)
+
+The user can select one of the many samples:
+
+![Project access tab with role select field to select an access role](samples.png)
+
+Currently, this opens a form to import a git repository. Other sample types aren't supported at the moment.
+
+## Motivation
+
+Creating a `ConsoleSample` CRD will promote the samples feature in the Developer Console to an extendable feature similar to QuickStarts. This makes it much easier for other teams to provide their own samples, for example with their operator.
+
+Samples are no longer tied to the exact OpenShift version and could be updated with the operator. Also customers can add their own samples if needed.
+
+This will also allow us to migrate from the (deprecated) [samples-operator](https://github.com/openshift/cluster-samples-operator) to a simple custom resource in the future and to add other sample types in the future.
+
+### User Stories
+
+* As an OpenShift Serverless engineer, I want to provide additional samples that are tied to an operator version, not an OpenShift release.
+* As a cluster administrator, I want to add additional samples to the Developer Console sample catalog.
+
+### Goals
+
+1. This feature should allow other teams to provide their Developer Console samples without applying them to the sample-operator or the console-operator.
+2. Samples should be allowed to localize/translated.
+
+### Non-Goals
+
+1. We don't want to turn off the support for the (deprecated) [samples-operator](https://github.com/openshift/cluster-samples-operator) right now. So the console will not provide any samples on its own (in 4.14).
+2. A way to hide installed samples. Operators creating a sample resource should have a way to opt out of their samples.
```

Comment:
We have customers who remove samples today by disabling the cluster samples operator, correct? Will they be annoyed that new samples pop up that they can't remove?

Reply:
That's correct. I discussed this with @JoelSpeed (and maybe with you as well), and he recommended that this should be an option in the providing operator.

So that the UI shows all `ConsoleSamples` and the Serverless operator must have an option to not re-create them again and again.

But you're right. That's different to ConsoleQuickStarts where we implemented this on our side... hmmm.... :thinking:

Reply:
As reference, this is our UI to disable Quick starts:

![image](https://github.com/openshift/enhancements/assets/139310/001e5ca1-623b-4fcd-9d6b-7f636ab3046a)

Reply:
The problem is that we're trusting operators to do that, and I worry from a UX standpoint that the experience of disabling samples for each operator will be different. It's something we can add later though, so I wouldn't hold the enhancement on that.